main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
> Also note that you are changing the wrong place as it is not going to work for fpm. Better solution would be to change php_syslog to function and do it in there.

@bukka Not sure I follow:

```
$ grep -r php_syslog
ext/standard/php_ext_syslog.h:#include "php_syslog.h"
ext/standard/syslog.c:	php_syslog(priority, "%s", message);
ext/standard/mail.c:#include "php_syslog.h"
ext/standard/mail.c:	php_syslog(LOG_NOTICE, "%s", message);
sapi/fpm/fpm/fpm_conf.c:#include "php_syslog.h"
sapi/fpm/fpm/fpm_stdio.c:#include "php_syslog.h"
sapi/fpm/fpm/zlog.c:#include "php_syslog.h"
sapi/fpm/fpm/zlog.c:		php_syslog(syslog_priorities[zlog_level], "%s", buf);
main/php_syslog.h:#define php_syslog std_syslog
main/php_syslog.h:#ifndef php_syslog
main/php_syslog.h:#define php_syslog syslog
main/php.h:#include "php_syslog.h"
main/main.c:#include "php_syslog.h"
main/main.c:					php_syslog(syslog_type_int, "%.*s", (int)valid_len, msg);
main/main.c:					php_syslog(syslog_type_int, "%.*s", (int)buf.len, buf.c);
$ 
```

`php_syslog()` is a macro pointing at either `syslog()` or else `std_syslog()`, and both are externals.